src/steps/validate-captcha.js

```diff
@@ -9,69 +9,58 @@
  * OF ANY KIND, either express or implied. See the License for the specific language
  * governing permissions and limitations under the License.
  */
-async function performValidation(fetch, token, secretKey) {
+
+import { PipelineStatusError } from '../PipelineStatusError.js';
+
+async function validateGoogleCaptchaV2(body, fetch, secretKey) {
+  const captchaToken = body.data?.['g-recaptcha-response']
+    || body.data?.find((x) => (x.name === 'g-recaptcha-response'))?.value;
+
+  if (!captchaToken) {
+    throw new PipelineStatusError(400, 'Captcha token missing from request body');
+  }
+
   const response = await fetch('https://www.google.com/recaptcha/api/siteverify', {
     method: 'POST',
-    body: new URLSearchParams({ secret: secretKey, response: token }),
+    body: new URLSearchParams({ secret: secretKey, response: captchaToken }),
   });
   if (!response.ok) {
-    return false;
+    throw new PipelineStatusError(response.status, `Captcha validation request returned ${response.status}`);
```

Comment:
I don't think that relying the response.status from google is a good idea. either use a `500` or `503`